xformers/components/attention/core.py

```diff
@@ -89,6 +89,8 @@ def _matmul_with_mask(
     # Non optimized codepath
     att = a @ b
     if mask.dtype == torch.bool:
+        if mask.ndim == 2:
```

Comment:
it does not work automatically given the broadcasting rules ?

Reply:
Yea I was surprised broadcasting wasn't working as well, but I was getting this error: 
<img width="753" alt="Screen Shot 2021-11-11 at 4 15 24 PM" src="https://user-images.githubusercontent.com/82468439/141370413-1ff1115a-2d2b-4928-b48d-90a24833a0f7.png">

Reply:
ahh, it's because we index explicitly here, so I'm guessing that broadcasting does not apply !